src/main.cpp

```diff
@@ -2216,6 +2217,7 @@ bool static ConnectTip(CValidationState &state, CBlockIndex *pindexNew, CBlock *
     // Remove conflicting transactions from the mempool.
     list<CTransaction> txConflicted;
     mempool.removeForBlock(pblock->vtx, pindexNew->nHeight, txConflicted, !IsInitialBlockDownload());
+    mempool.recalcPriorities(pindexNew->nHeight);
```

Comment:
I think we want to know priorities aimed for the next block's height, as that is where the transactions are expected to end up.